modules/metastore/modules/metastore_search/metastore_search.module

```diff
@@ -1,58 +1,66 @@
 <?php
 
 use Drupal\Core\Entity\EntityInterface;
+use Drupal\metastore\NodeWrapper\Data;
 
 /**
- * Manage item tracking of search api items.
+ * Determine whether the supplied entity is a dataset.
+ *
+ * @param \Drupal\Core\Entity\EntityInterface $entity
+ *   The entity in question.
+ *
+ * @return bool
+ *   Whether the entity is a dataset.
  */
-function metastore_search_search_api_tracking(EntityInterface $entity, string $operation) {
-  if ($entity->getEntityTypeId() != 'node') {
-    return;
-  }
-
-  if ($entity->bundle() != 'data') {
-    return;
-  }
-
-  $wrapper = new Drupal\metastore\NodeWrapper\Data($entity);
-  if ($wrapper->getDataType() != 'dataset') {
-    return;
-  }
-
-  $acceptableOperation = ['Inserted', 'Updated', 'Deleted'];
-
-  if (!in_array($operation, $acceptableOperation)) {
-    return;
-  }
-
-  $method = "trackItems{$operation}";
-
-  $storage = \Drupal::service('entity_type.manager')
-    ->getStorage('search_api_index');
-
-  /* @var $index \Drupal\search_api\Entity\Index */
-  $index = $storage->load('dkan');
-
-  $index->{$method}('dkan_dataset', [$entity->uuid()]);
+function _is_dataset(EntityInterface $entity): bool {
+  return $entity->getEntityTypeId() === 'node' && $entity->bundle() === 'data' && (new Data($entity))->getDataType() === 'dataset';
```

Comment:
@clayliddell isn't the third condition here all we need? We could wrap the `new Data()` call in a try/catch.

Reply:
Yeah, I agree. It's a bit less readable, but it at least guarantees consolidation of validation logic.